AnaProd/HH_bbWW/baseline.py

```diff
@@ -39,22 +39,25 @@ def RequestOnlyResolvedGenJets(df):
 
 
 def RecoHWWCandidateSelection(df):
-    df = df.Define("Electron_sel", """
-        v_ops::pt(Electron_p4) > 10 && abs(v_ops::eta(Electron_p4)) < 2.5 && abs(Electron_dz) < 0.2 && abs(Electron_dxy) < 0.045
-        && Electron_mvaIso_WP80 """) # this cut tells that electron is not inside of a jet
+    df = df.Define("Electron_presel", """
+        v_ops::pt(Electron_p4) > 10 && abs(v_ops::eta(Electron_p4)) < 2.5 && abs(Electron_dz) < 0.1 && abs(Electron_dxy) < 0.05 && Electron_sip3d <= 8 && Electron_miniPFRelIso_all < 0.4 && Electron_mvaIso_WP90""")
 
-    df = df.Define("Muon_sel", """
-        v_ops::pt(Muon_p4) > 15 && abs(v_ops::eta(Muon_p4)) < 2.4 && abs(Muon_dz) < 0.2 && abs(Muon_dxy) < 0.045
-        && Muon_tightId && Muon_pfRelIso04_all < 0.15""")
+    df = df.Define("Muon_presel", """
+        v_ops::pt(Muon_p4) > 15 && abs(v_ops::eta(Muon_p4)) < 2.4 && abs(Muon_dz) < 0.1 && abs(Muon_dxy) < 0.05 && abs(Muon_dxy) < 0.05 && Muon_sip3d <= 8 && Muon_miniPFRelIso_all < 0.4 && Muon_looseId""")
+
+    df = df.Define("Electron_sel", f"""
+        Electron_presel && Electron_mvaIso_WP80 """)
+
+    df = df.Define("Muon_sel", f"""
+        Muon_presel && ( (Muon_tightId && Muon_mediumPromptId ) || (Muon_highPtId && Muon_mediumPromptId ))""")
```

Comment:
doing AND or OR between different IDs is not recommended by Muon POG. At the level of anaTuple production we can have OR, but then we'll need to chose one ID for the final selection

Reply:
https://github.com/ahmad3213/Framework/commit/048c28a76dfcdb0e41bace605df22a6fbf2c3c38#diff-35aaf70173d771f9fcec0bb8cda6689dd87397e57ab7433e86e52e402c2a8f25R52

Reply:
updated to use Muon_tightId || Muon_highPtId